### PR TITLE
feat(learnxinyminutes): init

### PR DESCRIPTION
**Pull request by nikitarevenco**
## 🎉 Theme for Learn X in Y Minutes🎉

This website gives a tour of various programming languages so we can learn their syntax in like half an hour

image

## 💬 Additional Comments 💬

The dependency that they use to highlight their code blocks uses the same class in different contexts which makes it not possible to theme everything exactly how we want it to be themed. E.g. in typescript all operators and the `:` in either of these contexts:
```ts
let a: string = "foo";
const b = { c: "foo" };
```
Have the same class

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

**Comment by uncenter** [comment]
You'll need to update the `userstyles.yml` data file as per step 5 of https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md#creating-userstyles.

**Comment by nikitarevenco** [comment]
> You'll need to update the `userstyles.yml` data file as per step 5 of https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md#creating-userstyles.

Ok, I just did it

**Comment by nikitarevenco** [comment]
Fixed the failing checks

File: scripts/userstyles.yml
Comment (nikitarevenco) [reply scripts/userstyles.yml]: Oh yeah, that was a mistake. I just changed it to https://learnxinyminutes.com everywhere

**Comment by nikitarevenco** [comment]
I'm not entirely sure why this lint error is here, I carefully looked and it looks like both snippets are identical
image

**Comment by sgoudham** [comment]
The default accent was set to Sapphire instead of Mauve, I've updated it for you and it looks like the linter has found more errors @nikitarevenco

**Comment by nikitarevenco** [comment]
I'll remove all the `font-weight` properties but I fee

> The default accent was set to Sapphire instead of Mauve, I've updated it for you and it looks like the linter has found more errors @nikitarevenco

Thank you!

**Comment by nikitarevenco** [comment]
yay!
image

**Comment by sgoudham** [comment]
Just tried it out, looks really nice!

I just wanted to mention that over at [catppuccin/mdBook](http://mdbook.catppuccin.com/landing.html#code), we use `peach` foreground on a `crust` background for the inline code spans. I feel like this should also line up with that?

image

cc: @uncenter

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:238]: Please see https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments. You can import our existing port for the syntax highlighter used (Pygments) instead of theming it here.

**Comment by uncenter** [comment]
> Just tried it out, looks really nice!
> 
> I just wanted to mention that over at [catppuccin/mdBook](http://mdbook.catppuccin.com/landing.html#code), we use `peach` foreground for the inline code spans. I feel like this should also line up with that?
> 
> image
> 
> cc: @uncenter

I would think accent or text.

File: styles/learnxinyminutes/catppuccin.user.css
Comment (nikitarevenco) [reply styles/learnxinyminutes/catppuccin.user.css:238]: > Please see https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#pygments. You can import our existing port for the syntax highlighter used (Pygments) instead of theming it here.

I tried it but it doesn't look right

Before:

image


After:

image

My code editor with the same code: (i use catppuccin theme, with zero modifications)
image

**Comment by nikitarevenco** [comment]
> > Just tried it out, looks really nice!
> > I just wanted to mention that over at [catppuccin/mdBook](http://mdbook.catppuccin.com/landing.html#code), we use `peach` foreground for the inline code spans. I feel like this should also line up with that?
> > image
> > cc: @uncenter
> 
> I would think accent or text.

I think accent will be too distracting and text fits better, since that's basically what they use in their own theme

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [reply styles/learnxinyminutes/catppuccin.user.css:238]: That would be issues with the upstream port. Let's still go with it, and we can open an issue there. I'll review it later this week along with another port I need to review token highlighting for.

Comment (nikitarevenco) [reply styles/learnxinyminutes/catppuccin.user.css:238]: > That would be issues with the upstream port. Let's still go with it, and we can open an issue there. I'll review it later this week along with another port I need to review token highlighting for.

Where is the origin of that file on github so I can send a PR to fix? I have effectively created a version for pygments which aligns closer to what catppuccin really should look like, so I think a good starting point would be to merge my addition with what is already there

also - some considerations. Using `import` and sending an extra network request, is it really worth it? Why not just have people copy-paste everything into the userstyle. (since they'd be copy-pasting the css variables either way)

Comment (uncenter) [reply styles/learnxinyminutes/catppuccin.user.css:238]: > Where is the origin of that file on github so I can send a PR to fix? I have effectively created a version for pygments which aligns closer to what catppuccin really should look like, so I think a good starting point would be to merge my addition with what is already there

I've raised a tracking issue: https://github.com/catppuccin/python/issues/72. It's not as simple as copy pasting CSS class names, as it is generated differently.

> also - some considerations. Using `import` and sending an extra network request, is it really worth it? Why not just have people copy-paste everything into the userstyle. (since they'd be copy-pasting the css variables either way) otherwise each site which imports that file will load slightly slower

It is worth it because having so many different syntax highlighting implementations across all our (100) userstyles is infeasible. Nobody wants to update that. The network request is negligible on modern hardware, and we already require the latest versions of web browsers so it is a reasonable expectation.

**Comment by nikitarevenco** [comment]
Added the pygments import

File: scripts/userstyles.yml
Comment (uncenter) [inline scripts/userstyles.yml:456]:
```suggestion
    name: Learn X in Y Minutes
```

File: scripts/userstyles.yml
Comment (uncenter) [inline scripts/userstyles.yml:455]:
```suggestion
  learn-x-in-y-minutes:
```

Comment (uncenter) [reply scripts/userstyles.yml:455]: Make sure to change the name of the directory holding the userstyle and the metadata.

File: scripts/userstyles.yml
Comment (uncenter) [reply scripts/userstyles.yml:456]: Make sure to change the metadata..

File: scripts/userstyles.yml
Comment (uncenter) [inline scripts/userstyles.yml:458]:
```suggestion
    color: blue
```

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:105]:
```suggestion
    .share {
      background-color: @mantle;
      border-color: @crust;

      &:hover {
        background-color: lighten(@mantle, 10%);
      }
    }
```

Let's make sure to nest styles, like this. See https://lesscss.org/features/#parent-selectors-feature.

Comment (uncenter) [reply styles/learnxinyminutes/catppuccin.user.css:105]: Also, let's not use `lighten()` on a core color like mantle. Just use base or surface0, the next two shades.

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:115]: Nest here, same thing with `lighten()` on mantle.

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:131]: Same thing as above.

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:164]: Avoid changing non-color properties like this unless strictly required.
```suggestion
```

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/learnxinyminutes/catppuccin.user.css
Comment (nikitarevenco) [reply styles/learnxinyminutes/catppuccin.user.css]: I thought this looks better but I can see why it'd be best to have all userstyles *only* affect colors, otherwise we could argue for making these "tweaks" everywhere

File: styles/learnxinyminutes/catppuccin.user.css
Comment (nikitarevenco) [reply styles/learnxinyminutes/catppuccin.user.css]: added

**Comment by nikitarevenco** [comment]
I have renamed stuff

File: styles/learnxinyminutes/catppuccin.user.css
Comment (uncenter) [inline styles/learnxinyminutes/catppuccin.user.css:1]: Great work, thank you!

**Comment by NeonGamerBot-QK** [comment]
You should probably add styling for the 404 page.
<a href="https://learnxinyminutes.com/400">Screenshot 2024-09-03 at 2 03 53 PM
</a>
then theres a diffrent 404 page.
<a href="https://learnxinyminutes.com/404"> 
image
</a>
and then you should prob add 30% darken for visited urls on the main paige.

**Review by NeonGamerBot-QK — CHANGES_REQUESTED** [review]
prev comment

**Comment by nikitarevenco** [comment]
Added 15% darkening effect, since 30% looks like too much

**Comment by uncenter** [comment]
> Added 15% darkening effect, since 30% looks like too much

Nice, that's how it should be - 5 to 15% is the good range!

**Comment by nikitarevenco** [comment]
> You should probably add styling for the 404 page. [Screenshot 2024-09-03 at 2 03 53 PM ](https://learnxinyminutes.com/400) then theres a diffrent 404 page. [ image ](https://learnxinyminutes.com/404) and then you should prob add 30% darken for visited urls on the main page.

Their `<html>` element doesn't even take up the full height of the screen, if I style this page it would look really strange 

image

also, what's the url for the 2nd error page? i can't find it

**Comment by NeonGamerBot-QK** [comment]
> > You should probably add styling for the 404 page. [Screenshot 2024-09-03 at 2 03 53 PM ](https://learnxinyminutes.com/400) then theres a diffrent 404 page. [ image ](https://learnxinyminutes.com/404) and then you should prob add 30% darken for visited urls on the main page.
> 
> Their `<html>` element doesn't even take up the full height of the screen, if I style this page it would look really strange
> 
> image
> 
> also, what's the url for the 2nd error page? i can't find it

the other 404 page is https://learnxinyminutes.com/404 . i have no idea why the other 404 pages are different tbh but i feel like in the future they would use that (or they use it currently for deleted documentation)

**Comment by nikitarevenco** [comment]
> > > You should probably add styling for the 404 page. [Screenshot 2024-09-03 at 2 03 53 PM ](https://learnxinyminutes.com/400) then theres a diffrent 404 page. [ image ](https://learnxinyminutes.com/404) and then you should prob add 30% darken for visited urls on the main page.
> > 
> > 
> > Their `<html>` element doesn't even take up the full height of the screen, if I style this page it would look really strange
> > image
> > also, what's the url for the 2nd error page? i can't find it
> 
> the other 404 page is https://learnxinyminutes.com/404 . i have no idea why the other 404 pages are different tbh but i feel like in the future they would use that (or they use it currently for deleted documentation)

same issue here -
image


if i add colors they would only cover part of the screen, since i can't style where there isn't a `<html>` tag

**Comment by NeonGamerBot-QK** [comment]
> Their `<html>` element doesn't even take up the full height of the screen, if I style this page it would look really strange
i think you are able to maybe style the back by just styling the `html` tag, it works for me at least.

**Comment by nikitarevenco** [comment]
> > Their `<html>` element doesn't even take up the full height of the screen, if I style this page it would look really strange
> > i think you are able to maybe style the back by just styling the `html` tag, it works for me at least.

no idea why but for me the colors for either of the error pages don't show. i added this at the bottom:

```less
      
      html {
          background-color: @base !important !important;
      }
      
      p {
          color: @text !important !important;
      }
```

And i tried in both firefox and chrome, the entire page remains unstyled. Hmm, this is weird. are you not having an issue like this? I tried turning off my extensions and it didn't fix it

the other parts of the website work fine and update when i change the colors 🤔

**Comment by uncenter** [comment]
You can't use the `html` selector there because then you are trying to select `html html` - you already selected `html.light`/`html.dark` at the top of the userstyle where you use the mixin. Just get rid of the `html {}` part or change `html` to `&`.

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (NeonGamerBot-QK) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:27]: the 404 page has no html el with the light or dark class so it doesnt get added only. I think you would have to add some way to detect its the 404 page or add global styling im not sure.

**Comment by nikitarevenco** [comment]
image
image

Added theme for error pages

**Comment by nikitarevenco** [comment]
Fixed colors for the smiley face and text on the erro page

image
image

**Comment by NeonGamerBot-QK** [comment]
LGTM

**Review by NeonGamerBot-QK — APPROVED** [review]

**Comment by nikitarevenco** [comment]
styled "selected language"

before:
image


after:
image

**Comment by nikitarevenco** [comment]
Added darker color for visited & selected language

And also colored border

image

**Comment by uncenter** [comment]
Seems like you are still finding things to theme on your own. Lmk when you think it is complete by marking it as ready for review!

**Comment by nikitarevenco** [comment]
> Seems like you are still finding things to theme on your own. Lmk when you think it is complete by marking it as ready for review!

Oh, that was it I think!

**Review by NeonGamerBot-QK — APPROVED** [review]

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:1]: I'm unable to load this file.

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:93]: Pretty sure you need these to be `rgbify`-ed? See https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#highlightjs. Would be surprised if this works as is!

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:179]:
```suggestion
        background-color: @accent-color !important;
```
This might be causing the issue.

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (nikitarevenco) [reply styles/learn-x-in-y-minutes/catppuccin.user.css:93]: Hmm this is strange, all of the code is now colourless

image

This wasn't an issue before! It had color (although the color was wrong)

I tried placing the variables at the end, marking them all with important, it didn't work. And also tried the `#rgbify` mixin.

Comment (uncenter) [reply styles/learn-x-in-y-minutes/catppuccin.user.css:92]: It will work if you use the mixin! Copy the variables from that linked section, and add the rgbify mixin to the bottom of the userstyle.

Comment (uncenter) [reply styles/learn-x-in-y-minutes/catppuccin.user.css:92]: Oh wait no your variables are correct, you just used the wrong import link at the top. You used Highlight.js, it should be Pygments.

**Comment by nikitarevenco** [comment]
Ready now

**Comment by nikitarevenco** [comment]
Not sure what's the issue here
image


Line 100 literally just has `}` character

**Comment by isabelroses** [comment]
{601D16B9-5FDC-4795-8776-EBDECF36C036}
We seem to be distorting some links. [reproduction url](https://learnxinyminutes.com/docs/lua/)

**Comment by uncenter** [comment]
> Not sure what's the issue here image
> 
> Line 100 literally just has `}` character

Run `deno task lint:fix`.

**Comment by nikitarevenco** [comment]
> {601D16B9-5FDC-4795-8776-EBDECF36C036} We seem to be distorting some links. [reproduction url](https://learnxinyminutes.com/docs/lua/)

fixed

image

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:192]:
```suggestion
          background-color: darken(@accent-color, 5%) !important;
          color: @base !important;
          border-color: darken(@accent-color, 10%) !important;   
```

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:183]:
```suggestion
        color: darken(@accent-color, 5%) !important;
```

**Review by NeonGamerBot-QK — CHANGES_REQUESTED** [review]
update preview since lots of changes have happened.

**Comment by nikitarevenco** [comment]
> update preview since lots of changes have happened.

the syntax highlighting is still not accurate, that needs to be updated first. what I have on the preview is what it *should be*, but until https://github.com/catppuccin/python/issues/72 the syntax highlighting will remain incorrect

I can update the preview with incorrect syntax highlighting, or we can wait until that issue gets resolved before merging this. up to you

the rest of the stuff in the preview didn't change

**Comment by uncenter** [comment]
> update preview since lots of changes have happened.

None of the visible stuff really changed. No need.

**Comment by nikitarevenco** [comment]
> > update preview since lots of changes have happened.
> 
> None of the visible stuff really changed. No need.
> 
> > the syntax highlighting is still not accurate, that needs to be updated first. what I have on the preview is what it should be, but until [catppuccin/python#72](https://github.com/catppuccin/python/issues/72) the syntax highlighting will remain incorrect
> 
> Like I said, don't worry about that. The variables file will update when we get to it, this individual userstyle doesn't need to be concerned with it.

Okay 😄

**Comment by sgoudham** [comment]
> the syntax highlighting is still not accurate, that needs to be updated first. what I have on the preview is what it should be, but until https://github.com/catppuccin/python/issues/72 the syntax highlighting will remain incorrect

Myself, @backwardspy and @uncenter have been working on this for the past few hours and it should be much more accurate to the style guide, VSCode, JetBrains, etc where possible.

image

**Review by sgoudham — APPROVED** [review]
> None of the visible stuff really changed. No need.

Might be worth regenerating the previews since `false` is now `mauve` but I don't think it's too much of a concern.

It looks good to me, I'll let @uncenter approve and merge after their review.

**Comment by nikitarevenco** [comment]
> > the syntax highlighting is still not accurate, that needs to be updated first. what I have on the preview is what it should be, but until https://github.com/catppuccin/python/issues/72 the syntax highlighting will remain incorrect
> 
> Myself, @backwardspy and @uncenter have been working on this for the past few hours and it should be much more accurate to the style guide, VSCode, JetBrains, etc where the parser allows for it.
> 
> image

wow, this is very well done! 😃

**Review by isabelroses — APPROVED** [review]

File: scripts/userstyles.yml
Comment (uncenter) [inline scripts/userstyles.yml:464]:
```suggestion
    categories: [development, wiki]
```

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:65]: What in the world is going on with the formatting here 😅 can you format this file with Prettier or otherwise fix this?

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:122]:
```suggestion
    ::selection {
      background-color: fade(@accent-color, 30%);
    }

    input,
    textarea {
      &::placeholder {
        color: @subtext0 !important;
      }
    }
    
    .highlight {
          background-color: @mantle;
          pre {
              background-color: @mantle !important;
          }
      }

    &:not(.dark, .light) {
        p {
          color: @subtext0 !important;
        }
            
        span {
          color: @text;
        }
    }
```

File: styles/learn-x-in-y-minutes/catppuccin.user.css
Comment (uncenter) [inline styles/learn-x-in-y-minutes/catppuccin.user.css:168]:
```suggestion
      color: @text;
      border-bottom-color: @surface0;
```

**Review by uncenter — CHANGES_REQUESTED** [review]

**Review by uncenter — APPROVED** [review]

**Review by NeonGamerBot-QK — APPROVED** [review]

**Review by isabelroses — APPROVED** [review]

**Pull request merged**
